.githooks/pre-commit

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+
+ENTERPRISE_COMMIT_FILE=.enterprise-commit
+ENTERPRISE_REPO=./enterprise
+
+# Check enterprise repo is loaded
+if [ ! -d "$ENTERPRISE_REPO" ]; then
+  exit 0
+fi
+
+# check if enterprise repo has any unstaged changes
+ENTERPRISE_UNSTAGED=`cd $ENTERPRISE_REPO; git status -s -uall`
+if [ ! -z "$ENTERPRISE_UNSTAGED" ]; then
+  echo "ERROR: There are unstaged changes in enterprise repo. Please, commit and push them first."
+  echo "git status of $ENTERPRISE_RERO"
+  (cd $ENTERPRISE_REPO; git status)
+  exit 1
+fi
+
+# check if enterprise repo has no remote
+ENTERPRISE_BRANCH=`cd $ENTERPRISE_REPO; git name-rev --name-only HEAD`
+ENTERPRISE_REMOTE=`cd $ENTERPRISE_REPO; git config branch.$ENTERPRISE_BRANCH.remote`
+if [ -z "$ENTERPRISE_REMOTE" ]; then
+  echo "ERROR: Current enterprise repo branch '$ENTERPRISE_BRANCH' has no remote. Please, push your enterprise changes first."
+  exit 1
+fi
+
+ENTERPRISE_REMOTE_EXISTS=`cd $ENTERPRISE_REPO; git branch -a | grep remotes/$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH`
+if [ -z "$ENTERPRISE_REMOTE_EXISTS" ]; then
+  echo "ERROR: Enterprise repo remote branch '$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH' does not yet exist. Please, push your enterprise changes first."
+  exit 1
+fi
+
+# check if enterprise repo has unpushed commits
+ENTERPRISE_DIFF=`cd $ENTERPRISE_REPO; git rev-list --left-right $ENTERPRISE_BRANCH...$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH`
+
+if [ ! -z "$ENTERPRISE_DIFF" ]; then
+  echo "ERROR: There are either unpushed changes, or current branch '$ENTERPRISE_BRANCH' is behind '$ENTERPRISE_REMOTE/$ENTERPRISE_BRANCH'. Please, update your branch, commit and push your changes."
+  echo "Remote branch differences:"
+  echo $ENTERPRISE_DIFF
+  exit 1
+fi
+
+# Refresh and add the enterprise commit file. This should have no effect if the commit hasn't changed.
+STORED_COMMIT=`cat $ENTERPRISE_COMMIT_FILE`
+ENTERPRISE_COMMIT=`cd $ENTERPRISE_REPO; git rev-parse HEAD`
+
+echo $ENTERPRISE_COMMIT > $ENTERPRISE_COMMIT_FILE
+git add $ENTERPRISE_COMMIT_FILE
```

Comment:
We should silently update .enterprise-commit file like this. Consider for example, if I went into enterprise folder and checked out master branch for testing and forgot to switch it back to migrator-dev branch which it is pointing to right now. Then upon commit, .enterprise-commit file will be updated to point to master branch of enterprise repo, without knowledge of the developer. 

Maybe we can have an yes/no prompt? with detailed information of stored_commit, new commit, new commit's branch etc?

Reply:
I agree, it's better to make sure that this change is intentional, even if it adds an extra step.